go-controller/pkg/kubevirt/pod.go

```diff
@@ -1,8 +1,57 @@
 package kubevirt
 
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
 // AllowPodBridgeNetworkLiveMigration will return true if the pod belongs
 // to kubevirt and should use the live migration features
 func AllowPodBridgeNetworkLiveMigration(annotations map[string]string) bool {
 	_, ok := annotations[AllowPodBridgeNetworkLiveMigrationAnnotation]
 	return ok
 }
+
+// FindVMRelatedPods will return pods belong to the same vm annotated at pod
+func FindVMRelatedPods(client *factory.WatchFactory, pod *corev1.Pod) ([]*corev1.Pod, error) {
+	vmName, ok := pod.Labels[VMLabel]
+	if !ok {
+		return []*corev1.Pod{}, nil
+	}
+	vmPods, err := client.GetPodsBySelector(pod.Namespace, metav1.LabelSelector{MatchLabels: map[string]string{VMLabel: vmName}})
+	if err != nil {
+		return []*corev1.Pod{}, err
+	}
+	return vmPods, nil
+}
+
+// FindNetworkInfo will return the original switch name and the OVN pod
+// annotation from any other pod annotated with the same VM as pod
+func FindNetworkInfo(client *factory.WatchFactory, pod *corev1.Pod) (NetworkInfo, error) {
+	vmPods, err := FindVMRelatedPods(client, pod)
+	if err != nil {
+		return NetworkInfo{}, fmt.Errorf("failed finding related pods for pod %s/%s when looking for network info: %v", pod.Namespace, pod.Name, err)
+	}
+	networkInfo := NetworkInfo{
+		OriginalSwitchName: pod.Spec.NodeName,
+	}
+	for _, vmPod := range vmPods {
```

Comment:
I would expect something different; I would have expected you to grab all pods related to the VM, then sort them based on the creation timestamp, then get the "newest" one.

This implies you are calling the method with the "oldest" / original pod. How does the client know which one to use ?

Reply:
You don't need to find the oldest pod, you just need to find another one that is not you then grab NetworkInfo from it.

Reply:
This function from the context os a change at a pod, this is the pod that get passed and we find the NetworkInfor related to it at other similar pods.

Reply:
ovn-k reach this function with only the pod that is running the VM so `NodeName` is fine, the rest of the pods are used to copy some annotation that are the same for all of them related to the same VM.

Reply:
But I think you could make this code more robust rather than assuming all pods have the same annotations. Like:

```
for _, vmPod := range vmPods {
	firstVMPod := vmPod
	originalSwitchName, ok := firstVMPod.Labels[OriginalSwitchNameLabel]
	if !ok {
                continue
	}
	networkInfo.OriginalSwitchName = originalSwitchName

	status, ok := firstVMPod.Annotations[util.OvnPodAnnotationName]
	if !ok {
		continue
	}
	networkInfo.Status = status
        break
}

if len(networkInfo.Status) == 0 {
       return networkInfo, fmt.Errorf("missing %s annotation or label %s for vms related to pod %s/%s", util.OvnPodAnnotationName, OriginalSwitchNameLabel, pod.namespace, pod.name)
}
return networkInfo, nil

```